src/JMess.cpp

```diff
@@ -370,7 +370,7 @@ void JMess::disconnectAll()
          it != mConnectedPorts.end(); ++it) {
         OutputInput = *it;
 
-        if (jack_disconnect(mClient, OutputInput[0].toLatin1(), OutputInput[1].toLatin1())) {
+        if (jack_disconnect(mClient, OutputInput[0].toUtf8(), OutputInput[1].toUtf8())) {
```

Comment:
I think what probably happens is that there's an implicit .constData() being done on it, which should probably be made explicit. (Or going the std::string route would be another option.)  This seems to have been the style throughout JMess.cpp, so there's a few similar lines that it wouldn't hurt to change in the long run.

That said, the more problematic thing is that currently, we have a very scorched earth approach to the patching modes. We get rid of every connection to then rebuild everything from the ground up. Which is not so great if you have other Jack applications running or a generally more complicated set up. I've stopped using the auto-patching modes myself, and have my own jack client that I've developed to monitor ports in a smarter way (Once I get some time, I might be able to apply the approach that I'm taking there to the code here.)

Reply:
I second this. Let's remove patching from JackTrip and recommend a program that does just that in a nicer way. There's also [jack-matchmaker](https://github.com/SpotlightKid/jack-matchmaker) that looks very promising. But I haven't tried it yet. As I understand it, jack clients get notifications about new ports. If a new port matches a pattern jack-matchmaker connects this port. IMHO if we don't remove patching entirely from jacktrip, jacktrip should behave like this.

Switching to std::string would be nice especially in what will be the library part.

Reply:
I think the current patch modes definitely have their place, and can be useful in a lot of cases. We just need to manage them in a way that doesn't also interfere with unrelated jack clients. (And as a benefit, multiple hub servers on the one machine won't interfere with each other if they're using auto patching.)

Reply:
I agree that hub modes could be managed by a separate app. There are many different possible scenarious musicians might want and it's impossible to hardcode all of them. I'm using `jack-plumbing` with `--nojackportsconnect` option now. I haven't tested jack-matchmaker, it might be interesting as well. There is also a python jack binding that in theory might allow wrinting port connection logic in python scripts. But it all requires testing and developing reference solutions for standard cases before we can actually drop hubmodes support from jacktrip. In the short term I don't think it is a realistic option.

Reply:
In the meantime, I've just completely re-written the built in patcher so that it addresses the concerns I raised above - the patching modes will work as people are accustomed to, but without deleting any other connections that have been set up elsewhere. (And this implicit conversion issue is gone as well.)

Once this and the async hublistener pull request have been merged with dev I'll set up a pull request for it.

Reply:
Just want to +1 all the issues with hubpatch and also pile on that it scales very poorly. I've followed @antonrunov and am now just using --nojackportsconnect + jack-plumbing + jack_thru as an interim replacement.

Reply:
Yeah, even with small projects I've been using my own custom qt command line app because I haven't wanted to lose all of the existing connections every time a client joins. If you do get a chance (I know we're all pretty busy at the moment) it might be useful to test the qjacktrip branch to see if you come up against the same scalability issues with the replacement patcher code or whether that mitigates the problem.